lib/src/util/ascii_utils.dart

```diff
@@ -58,3 +58,18 @@ bool isValidDartFileName(String name) {
   }
   return true;
 }
+
+extension StringExtensions on String {
+  /// Returns whether `this` is just underscores.
+  bool get isJustUnderscores {
+    if (isEmpty) {
+      return false;
+    }
+    for (var i = 0; i < length; i++) {
+      if (!isUnderScore(codeUnitAt(i))) {
```

Comment:
I wonder whether it would be faster to test some common short versions explicitly:
```dart
switch (length) {
  case 1: return this == '_';
  case 2: return this -- '__';
  // ...
  default:
    for ...
}
```

Reply:
That implementation sounds great. Done.